cpl/module_cplfields.F90

```diff
@@ -0,0 +1,374 @@
+       "inst_merid_moment_flx                    ", &
+       "inst_sensi_heat_flx                      ", &
+       "inst_laten_heat_flx                      ", &
+       "inst_down_lw_flx                         ", &
+       "inst_down_sw_flx                         ", &
+       "inst_temp_height2m                       ", &
+       "inst_spec_humid_height2m                 ", &
+       "inst_zonal_wind_height10m                ", &
+       "inst_merid_wind_height10m                ", &
+       "inst_temp_height_surface                 ", &
+       "inst_pres_height_surface                 ", &
+       "inst_surface_height                      ", &
+       "mean_net_lw_flx                          ", &
+       "mean_net_sw_flx                          ", &
+       "inst_net_lw_flx                          ", &
+       "inst_net_sw_flx                          ", &
+       "mean_down_sw_ir_dir_flx                  ", &
+       "mean_down_sw_ir_dif_flx                  ", &
+       "mean_down_sw_vis_dir_flx                 ", &
+       "mean_down_sw_vis_dif_flx                 ", &
+       "inst_down_sw_ir_dir_flx                  ", &
+       "inst_down_sw_ir_dif_flx                  ", &
+       "inst_down_sw_vis_dir_flx                 ", &
+       "inst_down_sw_vis_dif_flx                 ", &
+       "mean_net_sw_ir_dir_flx                   ", &
+       "mean_net_sw_ir_dif_flx                   ", &
+       "mean_net_sw_vis_dir_flx                  ", &
+       "mean_net_sw_vis_dif_flx                  ", &
+       "inst_net_sw_ir_dir_flx                   ", &
+       "inst_net_sw_ir_dif_flx                   ", &
+       "inst_net_sw_vis_dir_flx                  ", &
+       "inst_net_sw_vis_dif_flx                  ", &
+       "inst_land_sea_mask                       ", &
+       "inst_temp_height_lowest                  ", &
+       "inst_spec_humid_height_lowest            ", &
+       "inst_zonal_wind_height_lowest            ", &
+       "inst_merid_wind_height_lowest            ", &
+       "inst_pres_height_lowest                  ", &
+       "inst_height_lowest                       ", &
+       "mean_fprec_rate                          ", &
+       "openwater_frac_in_atm                    "  &
+!      "northward_wind_neutral                   ", &
+!      "eastward_wind_neutral                    ", &
+!      "upward_wind_neutral                      ", &
+!      "temp_neutral                             ", &
+!      "O_Density                                ", &
+!      "O2_Density                               ", &
+!      "N2_Density                               ", &
+!      "height                                   "  &
+  /)
+  ! Field types should be provided for proper handling
+  ! according to the table below:
+  !  g : soil levels (3D)
+  !  i : interface (3D)
+  !  l : model levels (3D)
+  !  s : surface (2D)
+  !  t : tracers (4D)
+  character(len=*), public, parameter :: exportFieldTypes(NexportFields) = (/ &
```

Comment:
It is hard to associate the type and share of each field - consider array of structures to keep the name, type, and share state together.